platform/linux-generic/include/odp_internal.h

```diff
@@ -55,10 +56,13 @@ struct odp_global_data_s {
 	odp_cpumask_t control_cpus;
 	odp_cpumask_t worker_cpus;
 	int num_cpus_installed;
+	config_t libconfig; /*< Runtime config using libconfig */
+	uint8_t libconfig_enabled; /*< Runtime config enabled */
```

Comment:
`config_t libconfig`can always be initialized, but in case `ODP_CONF_FILE` is not given, no configuration file is read. I'm not sure if all libconfig functions handle this properly without crashing and I'm playing it safe here. Maybe it's cleaner if I replace this with:
```
	config_t *libconfig;
	config_t libconfig_structure;
```

Actually, this gave me an idea. It would probably make sense to save the `odp-linux.conf` contents to a C file during configure. This way we always have a valid default configuration and the default options have to be saved only to one location.

Reply:
As the last resort you can load it and then call `config_file_write()` from the helper app. I'd prefer instead if callers could provide default values fro options that they require.